src/keystore/index.ts

```diff
@@ -0,0 +1,18 @@
+import keystore from 'keystore-idb'
+import { KeyStore, CryptoSystem } from 'keystore-idb/types'
+
+const KEYSTORE_CFG = { type: CryptoSystem.RSA }
+
+let ks: KeyStore | null = null
+
+export async function get(): Promise<KeyStore> {
+  if(ks){
+    return ks
+  }
+  ks = await keystore.init(KEYSTORE_CFG)
+  return ks
+}
+
+export default {
+  get
+}
```

Comment:
Honest question: why not `export default function get()`

Also why not use an arrow functions in the vast majority of places? They're preferred now because of various simpler rules, no?

Reply:
I kind of like the semantics of 
```ts
import keystore from '../keystore'
const ks = await keystore.get()
```
But that's not a big deal. We could always something like this (which reads fine as well):
```ts
import getKeystore from '../keystore'
const ks = await getKeystore()
```

The bigger reason is that we might have a `set` function in this module as well (similar to the ipfs module). If a user wants to use their own predefined keystore, they can set it for the sdk. I just hadn't added that function yet :sweat_smile:

Reply:
As far as arrow functions go, isn't the main difference that they bind the current scope to the function? `.bind(this)`

I don't use `this` anywhere except for in methods on classes (where you wouldn't want to bind a larger scope than the class). 

I've just always done function declarations like this in files & used arrow functions for short anonymous functions in code. Happy to switch it up though if better/more reliable

Reply:
> I kind of like the semantics of

Cool sounds good 👍

Reply:
> Happy to switch it up though if better/more reliable

Yeah, there's way fewer edge cases (and/or is much more intuitive). `function` does behave in surprising ways sometimes that arrows don't. Implicit scoping can lead to some very subtle bugs.

An oldie but a goodie: https://stackoverflow.com/a/23045200

(I will note that `.bind(this)` is actually surprisingly slow, though I doubt it'll bite us in this case because we probably won't be calling it enough / often will pass explicitly as an arg)

But fundamentally this is your decision, given that you're more in the code.

Thoughts?

Reply:
Alright I'm convinced. I think I'll use arrow functions over function declarations when defining functions, but not for methods on classes. That seems like it can carry with it it's own class of bugs